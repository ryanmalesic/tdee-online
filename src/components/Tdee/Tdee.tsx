import React from 'react';

export interface TdeeProps {
  tdee: number;
}

const Tdee: React.FC<TdeeProps> = (props) => {
  const { tdee } = props;

  return (
    <div className="box">
      <h2 className="title is-2">Your TDEE</h2>
      <h4 className="subtitle is-4">Calculated with your recent logs</h4>
      <div className="container has-text-centered">
        <p className="subtitle is-3">
          {tdee ? (
            <strong className="has-text-primary">{tdee} kCals / Day</strong>
          ) : (
            'Not enough logs to calculate TDEE'
          )}
        </p>
      </div>
    </div>
  );
};

export default Tdee;
