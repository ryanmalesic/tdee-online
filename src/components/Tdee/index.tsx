import Tdee from './Tdee';

export { Tdee as default };
