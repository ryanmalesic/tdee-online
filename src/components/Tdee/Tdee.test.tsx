import React from 'react';
import renderer from 'react-test-renderer';

import Tdee from './Tdee';

it('renders Tdee unchanged', () => {
  const tree = renderer.create(<Tdee tdee={1234} />).toJSON();
  expect(tree).toMatchSnapshot();
});
