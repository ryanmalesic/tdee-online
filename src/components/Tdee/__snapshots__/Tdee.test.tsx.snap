// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tdee unchanged 1`] = `
<div
  className="box"
>
  <h2
    className="title is-2"
  >
    Your TDEE
  </h2>
  <h4
    className="subtitle is-4"
  >
    Calculated with your recent logs
  </h4>
  <div
    className="container has-text-centered"
  >
    <p
      className="subtitle is-3"
    >
      <strong
        className="has-text-primary"
      >
        1234
         kCals / Day
      </strong>
    </p>
  </div>
</div>
`;
